phoenix-core/src/main/java/org/apache/phoenix/util/ViewUtil.java

```diff
@@ -635,11 +639,52 @@ public static void addTagsToPutsForViewAlteredProperties(List<Mutation> tableMet
                         MetaDataEndpointImpl.VIEW_MODIFIED_PROPERTY_BYTES);
                 MetaDataUtil.conditionallyAddTagsToPutCells((Put)m,
                         PhoenixDatabaseMetaData.TABLE_FAMILY_BYTES,
-                        PhoenixDatabaseMetaData.VIEW_TTL_BYTES,
-                        parentViewTTLBytes,
+                        PhoenixDatabaseMetaData.PHOENIX_TTL_BYTES,
+                        parentPhoenixTTLBytes,
                         MetaDataEndpointImpl.VIEW_MODIFIED_PROPERTY_BYTES);
             }
 
         }
     }
+
+    public static ParseNode getViewWhereWithPhoenixTTL(StatementContext context, PTable table)
+            throws SQLException {
+        ParseNode viewWhere = null;
+        String viewStatement = null;
+
+        // If client side masking for PHOENIX_TTL is not enabled just return.
+        if (!ScanUtil.isClientSideMaskingEnabled(context.getConnection())) {
+            return viewWhere;
+        }
+
+        String viewTTLFilterClause = context.getConnection().getSCN() == null ?
+                String.format(
+                        "((TO_NUMBER(CURRENT_TIME()) - TO_NUMBER(phoenix_row_timestamp())) < %d)",
+                        table.getPhoenixTTL()) :
+                String.format(
+                "((%d - TO_NUMBER(phoenix_row_timestamp())) < %d)",
+                        context.getConnection().getSCN(), table.getPhoenixTTL());
+
+        if (table.getViewStatement() != null) {
```

Comment:
technically we are allowing the user to create a view with `SELECT * FROM BASE_TABLE` without anywhere clause. Do we want to count this case? 
Personally, I don't think anyone intentional to do this, but I will let @ChinmaySKulkarni to confirm we don't support this.

Reply:
The getViewStatement will be "not null" in the case when "SELECT * FROM BASE_TABLE" and the processing should proceed.